File: Python/214.py
class Solution(object):
    def shortestPalindrome(self, s):
        left = 0
        if s == s[::-1]:
            return s
        for right in range(len(s)-1,-1,-1):
            if s[left] == s[right]:
                left += 1
        return s[len(s)-1:left-1:-1] + self.shortestPalindrome(s[:left]) + s[left:]

File: Java/214.java
class Solution {
    public String shortestPalindrome(String s) {
        String rev = "", temp = "";
        int left = 0,right = 0;
        for(right = s.length() - 1; right > -1; right--){
            if(s.charAt(left) == s.charAt(right)) left++;
        }
        if(left == s.length()) return s;
        temp = s.substring(left);
        for(int i = temp.length() - 1; i > -1; i--) rev += temp.charAt(i);
        return rev + shortestPalindrome(s.substring(0,left)) + temp;
    }
}